components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -223,14 +235,25 @@ abstract class AbstractFetchDownloadService : Service() {
         val foregroundServiceId = downloadJobs[download.id]?.foregroundServiceId ?: Random.nextInt()
 
         // Create a new job and add it, with its downloadState to the map
-        downloadJobs[download.id] = DownloadJobState(
+        val downloadJobState = DownloadJobState(
             state = download,
             foregroundServiceId = foregroundServiceId,
-            status = DownloadJobStatus.ACTIVE
+            status = ACTIVE
         )
 
-        downloadJobs[download.id]?.job = CoroutineScope(IO).launch {
-            startDownloadJob(download.id)
+        downloadJobState.job = CoroutineScope(IO).launch {
+            startDownloadJob(downloadJobState)
+        }
+
+        setForegroundNotification(downloadJobState)
+
+        downloadJobs[download.id] = downloadJobState
+
+        // We are canceling the notificationUpdateScope, because we don't want to keep it
+        // running when we don't have more downloads to process. As a result, we need to
+        // create a new one when there are more downloads to process and the scope is canceled
+        if (!notificationUpdateScope.isActive) {
+            notificationUpdateScope = MainScope()
```

Comment:
I haven't read all the code yet, but here I do not understand why we need to create new scopes instead of having one scope that is tied to the lifetime of the service - because like the notification/service it should only be around if there's a download? 🤔

Reply:
That make sense, I will try to tied it to the service's lifecycle.

Reply:
Now `notificationUpdateScope` is tied to the same scope as the service 👍